code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,846 @@
+			spawn(5)
+				swapOccupants()
+				return TRUE
+		if(!occupantScan)
+			errorMessage = "Require scan"
+			return TRUE
+		if(currentlySwapping)
+			errorMessage = "Currently active: Please wait"
+			return TRUE
+	if(href_list["delay_swap"])
+		if(!currentlySwapping)
+			currentlySwapping = TRUE
+			spawn(5)
+				delayedSwap()
+				return TRUE
+
+//////Scan and Swap, other UI procs//////////////
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/unlockPods()
+	occupantScan = FALSE
+	mindTypeOne = "None"
+	mindTypeTwo = "None"
+	occupantNameOne = "None"
+	occupantNameTwo = "None"
+	occupantStatOne = "None"
+	occupantStatTwo = "None"
+	connectOne.icon_state = "mind_pod_open"
+	connectTwo.icon_state = "mind_pod_open"
+	playsound(connectTwo, 'sound/machines/door_unbolt.ogg', 35, 1)
+	playsound(connectOne, 'sound/machines/door_unbolt.ogg', 35, 1)
+	if(badSwap == FALSE)
+		flick("mind_pod_opening", connectOne)
+		flick("mind_pod_opening", connectTwo)
+	else
+		flick("mind_pod_thefly", connectOne)
+		flick("mind_pod_thefly", connectTwo)
+		badSwap = FALSE
+	lockedPods = FALSE
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/ejectPods()
+	connectOne.go_out()
+	connectTwo.go_out()
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/delayedSwap()
+	if(!occupantScan)
+		errorMessage = "Proceed with caution: supervisor advised"
+		sleep(5 SECONDS)
+		currentlySwapping = FALSE
+		if((occupantOne) && (occupantTwo))
+			scanOccupants()
+			if(prob(8-manipRating)) //A tiny bit risky with no upgrades, 0 risk with full xenoarch manipulators.
+				malfSwap = TRUE
+			swapOccupants()
+			unlockPods()
+		else
+			errorMessage = "Missing occupant"
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/scanOccupants(var/mob/S)
+	if((!locate(occupantOne,connectOne.contents)) || (!locate(occupantTwo, connectTwo.contents)))
+		errorMessage = "Missing occupant"
+		occupantOne = null
+		occupantTwo = null
+		return
+	currentlySwapping = TRUE
+	lockedPods = TRUE
+	if((connectOne.icon_state != "mind_pod_closed") && (connectTwo.icon_state != "mind_pod_closed"))
+		playsound(connectOne, 'sound/machines/poddoor.ogg', 55, 1)
+		playsound(connectTwo, 'sound/machines/poddoor.ogg', 55, 1)
+		flick("mind_pod_closing",connectOne)
+		connectOne.icon_state = "mind_pod_closed"
+		flick("mind_pod_closing",connectTwo)
+		connectTwo.icon_state = "mind_pod_closed"
+	sleep(2 SECONDS)
+	playsound(connectOne, 'sound/effects/sparks4.ogg', 80, 1)
+	playsound(connectTwo, 'sound/effects/sparks4.ogg', 80, 1)
+	occupantScan = TRUE
+	S = occupantOne
+	occupantNameOne = S.name
+	switch(S.stat)
+		if(0)
+			occupantStatOne = "Alive"
+		if(1)
+			occupantStatOne = "Unconscious"
+		if(2)
+			occupantStatOne = "Dead"
+	if(!S.mind)
+		mindTypeOne = "Lower" //Simple mob
+	if(S.mind)
+		if(isrobot(S))
+			mindTypeOne = "Artificial" //Silicon player, obviously
+		else
+			mindTypeOne = "Higher" //Player controlled
+	if(isvampire(S) || isanycultist(S) || ischangeling(S) || ismalf(S))
```

Comment:
!!!!! metatest inbound !!!!!

Reply:
Even the mind transfer wiz spell disallows those, they're completely borked, spent days trying to fix them but couldn't. I tried to make it a sort of thematic in-universe thing rather than meta "it don't work". 
In fairness every other check for them is easier than shoving them in a mind machine.

Reply:
Maybe just make them go through with the transfer as normal but then fail with the brain damage thing in the end.

Reply:
I can change it if you want but consider that this requires moderately expensive construction, very demanding research, and 3 machines to be anchored adjacent to each other. It doesn't require a bluespace crystal to scan but if they're not intentionally tossing them in there it would be required to accidentally discover one. The alternative is drag to church, use any chaplain item, or implant. 2 of which are portable. I just don't think it's a viable test at all and it might even lead to a fun exchange where a cultist has to talk their way out of entering the mind machine without raising suspicion.